l2geth/rollup/client.go

```diff
@@ -104,13 +102,13 @@ type signature struct {
 // When this struct exists in other structs and is set to `nil`,
 // it means that the decoding failed.
 type decoded struct {
-	Signature signature      `json:"sig"`
-	Value     hexutil.Uint64 `json:"value"`
-	GasLimit  uint64         `json:"gasLimit"`
-	GasPrice  uint64         `json:"gasPrice"`
-	Nonce     uint64         `json:"nonce"`
-	Target    common.Address `json:"target"`
-	Data      hexutil.Bytes  `json:"data"`
+	Signature signature       `json:"sig"`
+	Value     hexutil.Uint64  `json:"value"`
+	GasLimit  uint64          `json:"gasLimit"`
+	GasPrice  uint64          `json:"gasPrice"`
+	Nonce     uint64          `json:"nonce"`
+	Target    *common.Address `json:"target"`
```

Comment:
does switching this to a pointer functionally change anything?

Reply:
Yes - it allows for it to be `nil`. The `nil` target means that its a contract creation. Previously we used a custom serialization that didn't allow for the encoding of `nil` so we used `address(0)` to mean address creation. Now that we are using RLP encoded transactions in the batches, we can properly encode `nil` so this change basically makes the batch deserialization logic match how normal ethereum transactions work